digitalpalireader/content/js/dict_xml.js

```diff
@@ -1,796 +1,634 @@
-      (sn ? ' (' + G_nikLongName[nikaya] + ' ' + sn + ')' : '')
-
-    finout +=
-      '<p>' +
-      placen +
-      ' <span class="abut obut" onclick="openPlace([\'' +
-      nikaya +
-      "','" +
-      bookno +
-      "','" +
-      pca[2] +
-      "','" +
-      pca[3] +
-      "','" +
-      pca[4] +
-      "','" +
-      pca[5] +
-      "','" +
-      pca[6] +
-      "','" +
-      hiert +
-      '\'],null,null,eventSend(event));">go</span></p>'
-  }
-  displayDictData(finout, 0);
-  if (!DPR_PAL.isXUL) {
-    document.getElementById('paliTextContent').scrollTop = 0;
-  }
+    displayDictData(finout);
+		if (!DPR_PAL.isXUL) {
+			document.getElementById('paliTextContent').scrollTop = 0;
+		}
 }
 
-function getDppnData (link) {
-  var dppnf = DPR_PAL.contentFolder + 'etc/XML2/' + link.split('/')[0] + '.xml'
-
-  var xmlhttp = new window.XMLHttpRequest()
-  xmlhttp.open('GET', dppnf, false)
-  xmlhttp.send(null)
-  var xmlDoc = xmlhttp.responseXML.documentElement
-
-  var data =
-    ' ' +
-    xmlDoc
-      .getElementsByTagName('e')
-      [parseInt(link.split('/')[1])].textContent.replace(/\[/g, '<')
-      .replace(/\]/g, '>')
-      .replace(/href/g, 'style="color:blue" href')
-      .replace(/\.  /g, '.&nbsp; ')
-  return data
+
+
+function getDppnData(link){
+	var dppnf = DPR_PAL.contentFolder + 'etc/XML2/'+link.split('/')[0]+'.xml';
+
+	var xmlhttp = new window.XMLHttpRequest();
+	xmlhttp.open("GET", dppnf, false);
+	xmlhttp.send(null);
+	var xmlDoc = xmlhttp.responseXML.documentElement;
+
+	var data = ' ' + xmlDoc.getElementsByTagName('e')[parseInt(link.split('/')[1])].textContent.replace(/\[/g, '<').replace(/\]/g, '>').replace(/href/g, 'style="color:blue" href').replace(/\.  /g, '.&nbsp; ');
+	return data;
 }
 
-function displayDictData (data, context) {
-  if (DPR_PAL.isXUL) {
-    var dataNode = document.createElement('div')
-    dataNode.innerHTML = data
-    document.getElementById('difb').setAttribute('align', 'left')
-    document.getElementById('difb').appendChild(dataNode)
-    document.getElementById('cdif').scrollTop = 0
-  } else {
-    var dataNode = $('<div></div>').html(data)
-    switch(context) {
-      case 0: //in the mainpane
-        $('#difb').html('')
-        $('#difb').append(dataNode)
-      break;
-      case 1: //in the bottompane
-        $('#difb2').html('')
-        $('#difb2').append(dataNode)
-      break;
-    }
-    $('#cdif').scrollTop(0)
-  }
+function displayDictData(data) {
+	if (DPR_PAL.isXUL) {
+		var dataNode = document.createElement('div');
+		dataNode.innerHTML = data;
+		document.getElementById('difb').setAttribute('align','left');
+		document.getElementById('difb').appendChild(dataNode);
+		document.getElementById('cdif').scrollTop=0;
+	} else {
+    const difbId = `#${DPR_PAL.getDifId()}`;
+		var dataNode = $('<div></div>').html(data);
+		$(difbId).html('');
+		$(difbId).append(dataNode);
+		$('#cdif').scrollTop(0);
```

Comment:
```suggestion
		$('#paliTextContent').scrollTop(0);
```

Reply:
thanks for catching this. i'll make this dynamic as well.

right after alpha - i want to discuss with you my ideas for modernizing the code base. one of which is to do away with the hardcoded html IDs.